src/main/java/io/cdap/plugin/gcp/bigquery/util/BigQueryDataParser.java

```diff
@@ -61,49 +62,97 @@ private BigQueryDataParser() {
   }
 
   public static StructuredRecord getStructuredRecord(Schema schema, FieldList fields, FieldValueList fieldValues) {
+
     StructuredRecord.Builder recordBuilder = StructuredRecord.builder(schema);
-    for (Field field : fields) {
+    List<Schema.Field> schemaFieldList = schema.getFields();
+
+    for (int i = 0; i < fields.size(); i++) {
+      Field field = fields.get(i);
       String fieldName = field.getName();
       FieldValue fieldValue = fieldValues.get(fieldName);
-      FieldValue.Attribute attribute = fieldValue.getAttribute();
+      Attribute attribute = fieldValue.getAttribute();
       if (fieldValue.isNull()) {
         recordBuilder.set(fieldName, null);
         continue;
       }
 
-      if (attribute == FieldValue.Attribute.REPEATED) {
+      Schema.Field localSchemaField = schemaFieldList.get(i);
```

Comment:
Are you assuming CDAP schema and BQ schema has same fields in same order?

Reply:
Refactored to avoid the assumption.